File: src/main/java/frc/team449/control/holonomic/HolonomicDrive.java
package frc.team449.control.holonomic;

import frc.team449.control.DriveSubsystem;

public interface HolonomicDrive extends DriveSubsystem {
  /**
   * The max speed that this drivetrain can translate at, in meters per second
   */
  double getMaxLinearSpeed();

  /**
   * The max speed that this drivetrain can turn in place at, in radians per second
   */
  double getMaxRotSpeed();
}

File: src/main/kotlin/frc/team449/control/holonomic/HolonomicDrive.kt
package frc.team449.control.holonomic

import frc.team449.control.DriveSubsystem

interface HolonomicDrive : DriveSubsystem {
  /**
   * The max speed that this drivetrain can translate at, in meters per second
   */
  val maxLinearSpeed: Double

  /**
   * The max speed that this drivetrain can turn in place at, in radians per second
   */
  val maxRotSpeed: Double
}
